### PR TITLE
Flush messages on close

### PR DESCRIPTION
**Pull request by richardnagle**
This PR will ensure that on appender shutdown all queued messages are sent to loggly.

### CODE REVIEW

**Comment by mostlyjason** [comment]
Thanks @richardnagle! We'll review this when our engineer gets a chance.

**Comment by abedra** [comment]
@mostlyjason when do you expect to merge this?

**Comment by YouGitMoreWithLes** [comment]
I would really like to see this get merged and released. I have a few console applications using this dependency and I must add a wait loop until the timer fires to flush the logs.

File: source/log4net-loggly/LogglyAsyncHandler.cs [hunk 1]
```diff
@@ -52,6 +53,14 @@ public void PostMessage(string msg, ILogglyAppenderConfig config)
 				Queue.TryAdd(msg);
 			}
 		}
+
+	    public void Flush()
+	    {
+	        while (IsRunning && Queue.Any())
```
Comment (Katulus) [inline source/log4net-loggly/LogglyAsyncHandler.cs]: This does not guarantee that messages from queue are really sent. Message is taken from queue before it's sent so this condition can become `true` but messages are still not sent. The application can terminate before the actual HTTP request is made and data are delivered to Loggly.

**Comment by Katulus** [comment]
The current implementation still does not guarantee that all logs are sent. Because messages are taken out of queue before they are sent over the wire the `Flush` call can return before logs are really sent over the wire. If you then terminate application you lose logs.

We plan to do some refactoring of this so we'll try to include proper flush on shutdown.

**Comment by Katulus** [comment]
New version (beta) has been just released. You can get it on https://github.com/loggly/log4net-loggly/releases/tag/v9.0.0-beta1 or via Nuget. It's almost complete rewrite and flush-on-shutdown is now included.

**Pull request closed without merging**